__tests__/app.test.js

```diff
@@ -0,0 +1,45 @@
+const testData = require('../db/data/test-data/index')
+const testDb = require('../db/connection')
+const seed = require('../db/seeds/seed')
+const app = require('../app')
+const request = require('supertest')
+
+beforeEach(() => seed(testData))
+afterAll(() => testDb.end)
+
+describe('GET/api/topics/invalidinput', () => {
+    test('404: responds with a message when an invalid endpoint has been requested', () => {
+        return request(app)
+        .get('/api/topics/invalidinput')
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe('not found')
+        })
+    })
+})
+
+describe('GET/api/topics', () => {
+    test('200: should return with an array of topic objects, with the correct properties', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body}) => {
+            const topics = body.topics
+            topics.forEach((topic) => {
+                expect(topic).toHaveProperty('slug')
+                expect(topic).toHaveProperty('description')
```

Comment:
Nice use of a `forEach` here, two things to consider with this however:
- if the array is empty the loop will be skipped, meaning lines 29 & 30 won't run and the test will default to a pass, a check for length outside the loop would protect against that
- You could be more robust in the testing, rather than `toHaveProperty` consider using `typeof` to check what type of data `topic.slug` is, that way you're testing both that the property `slug` exists and asserting that is should be a `string` ( same for description )